components/style/properties/properties.mako.rs

```diff
@@ -3336,10 +3314,42 @@ where
                 continue;
             }
 
+            if
+                % if category_to_cascade_now == "early":
+                    !
+                % endif
+                longhand_id.is_early_property()
+            {
+                continue
+            }
+
+            <% maybe_to_physical = ".to_physical(writing_mode)" if category_to_cascade_now != "early" else "" %>
+            let physical_longhand_id = longhand_id ${maybe_to_physical};
+            if seen.contains(physical_longhand_id) {
+                continue
+            }
+            seen.insert(physical_longhand_id);
+
+            let mut resolved_declaration = match *declaration {
+                PropertyDeclaration::WithVariables(id, ref unparsed) => {
+                    if !id.inherited() {
+                        context.rule_cache_conditions.borrow_mut()
+                            .set_uncacheable();
+                    }
+                    Cow::Owned(unparsed.substitute_variables(
+                        id,
+                        context.builder.custom_properties.as_ref(),
+                        context.quirks_mode
+                    ))
+                }
+                ref d => Cow::Borrowed(d)
+            };
+
+
             // When document colors are disabled, skip properties that are
             // marked as ignored in that mode, if they come from a UA or
```

Comment:
(Unrelated to this change, but I guess the "if" here should actually be "unless".)

Reply:
Yup, agreed, fixed.